internal/core/encoding.go

```diff
@@ -288,3 +288,79 @@ func EncodeIndexDataStoreKey(key *IndexDataStoreKey) []byte {
 
 	return b
 }
+
+func DecodeDataStoreKey(data []byte) (DataStoreKey, error) {
+	if len(data) == 0 {
+		return DataStoreKey{}, ErrEmptyKey
+	}
+
+	if data[0] != '/' {
+		return DataStoreKey{}, ErrInvalidKey
+	}
+	data = data[1:]
+
+	data, colRootID, err := encoding.DecodeUvarintAscending(data)
+	if err != nil {
+		return DataStoreKey{}, err
+	}
+
+	var instanceType InstanceType
+	if len(data) > 1 {
```

Comment:
suggestion: wouldn't it be better to do:
```go
if len(data) <= 1 {
  return ...
}

// check instance type ...
```
Currently if `if len(data) > 1` is false the rest of the function is irrelevant, but is being executed.

Reply:
I do prefer the simplicity of the current code, although it does have a small runtime cost. I'll have a quick look when I remove the magic number - do let me know if you have a strong preference though.

- [ ] Have quick look at early returns

Reply:
I wouldn't call it a strong preference. Just following return-early best practice.

Reply:
I had a play, whilst I like it hightlights the data flow a little more nicely, it ends up duplicating logic (or mutating, or a new func) and even without my gut prefers the feel of the current. Leaving as-is.